### PR TITLE
Adding code coverage for pkg/bundle

### PR DESCRIPTION
**Pull request by junshun**
Closes: #164 

*Description of changes:*
Unit testing was added for IsBundleKnown and LatestBundle for the bundle manager.

Unit Test was created for client with mock interfaces added

Coverage is now at 92% for pkg/bundle. Most of the remaining percentage is in the mock code for client.

### CODE REVIEW

File: pkg/bundle/client_test.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	})
+}
+
+// Helpers
+type mockClient struct {
+	client.Client
+	err error
+}
+
+func newMockClient(err error) *mockClient {
```
Comment (TerryHowe) [inline pkg/bundle/client_test.go:133]: Can this be thrown in this https://github.com/aws/eks-anywhere-packages/blob/main/Makefile#L163

Comment (junshun) [reply pkg/bundle/client_test.go]: Addressed in latest commit. Used the mock client interface that was already generated in controllers/mocks

File: pkg/bundle/client_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+		namespacedNames, err := bundleClient.GetActiveBundleNamespacedName(ctx)
+
+		assert.Equal(t, api.PackageNamespace, namespacedNames.Namespace)
+		assert.Equal(t, "", namespacedNames.Name)
+		assert.Nil(t, err)
+	})
+}
+
+// Helpers
```
Comment (TerryHowe) [inline pkg/bundle/client_test.go:138]: My minor nit on these helper names, for tests, I normally call these something like `givenMockClient` so they don't show up when someone is trying to remember what the name of some new function. Also given follows the testing given/when/then pattern. We are putting tests in the same package as our code and the danger with doing that is someone will accidentally use test code in a production implementation.

**Review by TerryHowe — APPROVED** [review]
/lgtm
/approve

**Pull request merged**